text/metrics/0113-exemplars.md

```diff
@@ -0,0 +1,104 @@
+
+  // When sample_count is non-zero, this exemplar has been chosen in a statistically
+  // unbiased way such that the exemplar is representative of `sample_count` individual events
+  optional double sample_count = 6;
+}
+```
+
+Exemplar collection should be enabled through an optional parameter (disabled by default), and when not enabled, there should be no collection/logic performed related to exemplars. This is to ensure that when necessary, aggregators are as high performance as possible. Aggregators should also have a parameter to determine whether exemplars should only be collected if they are recorded during a sampled trace, or if tracing should have no effect on which exemplars are sampled. This allows aggregations to prioritize either the link between metrics and traces or the statistical significance of exemplars, when necessary.
+
+[#347](https://github.com/open-telemetry/opentelemetry-specification/pull/347) describes a set of standard aggregators in the metrics SDK. Here we describe how exemplars could be implemented for each aggregator.
+
+### Exemplar behaviour for standard aggregators
+
+#### HistogramAggregator
+
+The HistogramAggregator MUST (when enabled) maintain a list of exemplars whose values are distributed across all buckets of the histogram (there should be one or more exemplars in every bucket that has a population of at least one sample-able measurement). Implementations SHOULD NOT retain an unbounded number of exemplars.
+
+#### Sketch
+
+A Sketch aggregator SHOULD maintain a list of exemplars whose values are spaced out across the distribution. There is no specific number of exemplars that should be retained (although the amount SHOULD NOT be unbounded), but the implementation SHOULD pick exemplars that represent as much of the distribution as possible. (Specific details not defined, see open questions.)
+
+#### Last-Value
+
+Most (if not all) Last-Value aggregators operate asynchronously and do not ever interact with context. Since the value of a Last-Value is the last measurement (essentially the other parts of an exemplar), exemplars are not worth implementing for Last-Value.
+
+#### Exact
+
+The Exact aggregator will function by maintaining a list of `RawValue`s, which contain all of the information exemplars would carry. Therefore the Exact aggregator will not need to maintain any exemplars.
+
+#### Counter
+
+Exemplars give value to counter aggregations in two ways: One, by tying metric and trace data together, and two, by providing necessary information to re-create the input distribution. When enabled, the aggregator will retain a bounded list of exemplars at each checkpoint, sampled from across the distribution of the data. Exemplars should be sampled in a statistically significant way.
+
+#### MinMaxSumCount
+
+Similar to Counter, MinMaxSumCount should retain a bounded list of exemplars that were sampled from across the input distribution in a statistically significant way.
```

Comment:
Would it make sense to always retain an exemplar for the min, and the max?

Reply:
Keeping the min & max was the original definition when I first made the OTEP 😬 At some point I changed it to this definition for the sake of allowing the exemplars to be used statistically (consistent with other aggregators), which min + max would not work with.

I think a common theme that I've been seeing is there is two uses of exemplars that benefit from different ways of collecting them; one is viewing singular exemplars manually where we prefer having traces attached and exemplars more relevant to the aggregator, and the other is using a larger group of exemplars to determine information about the input distribution, common tags, etc, which requires statistically sampled exemplars. Maybe there is a way to satisfy both sides with one definition for aggregators like MMSC, but I can't think of it.

Reply:
Heh. ok. I'm totally fine with this for now...it was just an idea (and obviously one that had already been considered!).  👍

Reply:
This is a fine point. (I'm still thinking about this.)

Reply:
OK, I have thoughts. In https://github.com/open-telemetry/opentelemetry-specification/issues/636 I've begun to propose that the default aggregator ValueRecorder become MinMaxLastSumCount, which I think of as being "like" 3 non-probabilistic exemplars (all extreme values) and two aggregations.  This could be represented as 5 aggregations, in the inexpensive case. It could be replaced by 2 aggregations and 3 exemplars. These would be the standard output of the instrument, not considering any additional randomly selected exemplars, which would have probability values associated with them.

Reply:
Therefore: I think this is a protocol question. I think it would be nice if we had the option to include full tags in "exemplar form" for MIN/MAX/LAST or not, depending on configuration, because the protocol made it so.